client/widmgr/widmgr.go

```diff
@@ -54,10 +54,23 @@ type WIDMgr struct {
 func NewWIDMgr(signer IdentitySigner, a *structs.Allocation, logger hclog.Logger) *WIDMgr {
 	widspecs := map[string][]*structs.WorkloadIdentity{}
```

Comment:
Could we use the `WIHandle` as the map key directly? Especially given that the `lastToken` map is keyed by the `WIHandle` it seems like that might make this easier to reason about.